### PR TITLE
Clickable Tucker Island Map #24

### PR DESCRIPTION
**Pull request by CodeWithTheDoctor**
## Change Summary
- Towns can be clicked
  - This will force the screen to zoom into the town location
- Dialouge box appears when a town is clicked.
- Supports panning
- On touch screen devices, "pinching" works. (I.e. can use two fingers to zoom in and out)

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

**Review by chenga3 — APPROVED** [review]
https://user-images.githubusercontent.com/39249792/151640991-872df774-66f5-4e3c-a6e8-2fca475e4b0a.mov

Found this bug where if you double click anywhere that's not a town, then the towns can't be clicked into. This is fairly minor though, can be raised as a separate issue and fixed in a different PR.

File: components/Map/assets/description.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
+{
```
[the remark this answers is not in the document]
Comment (CodeWithTheDoctor) [reply components/Map/assets/description.json:1]: > Screen.Recording.2022-01-29.at.9.08.45.am.mov 
> Found this bug where if you double click anywhere that's not a town, then the towns can't be clicked into. This is fairly minor though, can be raised as a separate issue and fixed in a different PR.

Huh strange, I just tried doing that on my device (Windows 10, Edge browser) and it was working as expected.

What device and browser did you try that on?

[the remark this answers is not in the document]
Comment (CodeWithTheDoctor) [reply components/Map/assets/description.json:1]: > Doesn't have to be done in this PR but I'm guessing we'll want to store this info + location images in the Notion DB? Or at least the stuff that the client may want to modify (eg. captionText + headerText)

Very low priority at the moment, but yeah this is a great feature to consider.

[the remark this answers is not in the document]
Comment (chenga3) [reply components/Map/assets/description.json:1]: Hmm yeah seems like it works on Chrome for me. Safari is where I found this bug.

**Comment by APM246** [comment]
Do better

**Comment by CodeWithTheDoctor** [comment]
- 10 Hours.

**Review by CodeWithTheDoctor — COMMENTED** [review]
Make the required changes :P

File: components/Map/MapImage.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  scale: number
+  initialScale: number
+  setTransform: (xtrans: number, ytrans: number, scale: number) => void
+  windowDimensions: {
+    height: number
+    width: number
+  }
+  setIsShowing: (state: boolean) => void
+  isShowing: boolean
+  selected: any
```
[the remark this answers is not in the document]
Comment (skyheat) [reply components/Map/MapImage.tsx]: Leaving it as any as it can be null

File: components/Map/MapImage.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            } else if (aspectRatio > 0.65) {
+              zombieAquaCompensation = 40 * initialScale
+            } else {
+              mobileCompensation = 350
+              zombieAquaCompensation = 200
+            }
+
+            const xtrans =
+              parseInt(location.xtrans) * 12 * scale * initialScale +
+              mobileCompensation
```
Comment (skyheat) [inline components/Map/MapImage.tsx:78]: Multiplied by 12 because xtrans and ytrans are values which are scaled down by 10-12

**Review by chenga3 — APPROVED** [review]

**Review by chenga3 — APPROVED** [review]

**Pull request merged**